source/GameInterface/Services/TroopRosters/Patches/TroopRosterLifetimePatch.cs

```diff
@@ -0,0 +1,65 @@
+﻿using Common.Logging;
+using Common.Messaging;
+using GameInterface.Policies;
+using GameInterface.Services.Armies.Messages.Lifetime;
+using GameInterface.Services.TroopRosters.Messages;
+using HarmonyLib;
+using Serilog;
+using System;
+using TaleWorlds.CampaignSystem.Party;
+using TaleWorlds.CampaignSystem.Roster;
+using TaleWorlds.CampaignSystem.Settlements.Buildings;
+
+namespace GameInterface.Services.TroopRosters.Patches
+{
+    /// <summary>
+    /// Lifetime Patches for TroopRoster
+    /// </summary>
+    [HarmonyPatch]
+    internal class TroopRosterLifetimePatches
+    {
+        private static ILogger Logger = LogManager.GetLogger<TroopRosterLifetimePatches>();
+
+        [HarmonyPatch(typeof(TroopRoster), MethodType.Constructor, typeof(PartyBase))]
```

Comment:
If ownerParty property is synced we dont need this patch, since then ownerParty should be patched whenever it´s constructor is called.
Since this is a chained ctor, it will call the ctor without the parameter first. So you could just use property patch to set the ownerParty property in the "2nd ctor".

Reply:
You are correct, I missed this. Good catch.